### ISSUE TITLE
Cannot open context menu to see the conversion actions

### ISSUE THREAD

**Issue by vfonic** [wix/vscode-glean#84]
I'm having some difficulties opening the context menu with the glean actions. If I right-click on the editor side, nothing opens. The only way I can use glean is if, sometimes glean actions present themselves in a "lightbulb's" context menu.

I don't know how to best explain this, so here's a screen recording:
https://www.dropbox.com/s/48jofoz4qlt9srp/Glean.mov?dl=0

**Comment by jacquesdev** [comment]
Same issue here

**Comment by borislit** [comment]
Hey @jacquesdev  @vfonic. Basically, glean actions are only available through the lighbulb context menu, as this is the vscode standard for context based actions (i.e make something available only when relevant). 
Do you have issues with having the actions appear in the lightbulb? 
Can you please post the code snippet u are trying to refactor? I'd be glad to have a look

**Comment by vfonic** [comment]
Hey @borislit!

The issue is that the lightbulb doesn't appear at all. Not the action in the lightbulb, but the lightbulb itself.

Glean doesn't trigger the lightbulb to appear. I can only use glean action when some other extension creates the lightbulb. Then, I _do_ see the glean action.

My current approach is to misspell some word and use Code Spell Checker extension, which creates lightbulb, to open lightbulb context menu and choose glean action.

**Comment by aeciorc** [comment]
@vfonic I've noticed that if there are multiple actions and one of them is preferred, the lightbulb appears blue and near the line of code corresponding to that action, rather than above the selected chunk of code. At first this made me think that it was not appearing at all. Maybe the same thing is happening in your case?

**Comment by vfonic** [comment]
@aeciorc no, I'm pretty sure the lightbulb sometimes doesn't appear at all.

I usually see (blue or yellow) lightbulb only if there's something else that can open that context menu. Are you sure the same is not happening in your case?

**Comment by borislit** [comment]
guys just a short update. Im on a short vacation. Ill try to help in a week or so when im back

**Comment by yyynnn** [comment]
Not working for me too

**Comment by borislit** [comment]
@yyynnn this is aligned with the rest of VSCode. Code actions that are context based (i.e appear when relevant) should be in the lightbulb menu. Code pallet commands appear no matter what.

Guys, @vfonic @aeciorc Im back and checking the issue

**Comment by vfonic** [comment]
Thank you, sir!

It works with the lightbulb. But it seems like Glean doesn't trigger the lightbulb appearance. I think that's the main issue. So, if there's something else triggering the lightbulb, I can get the action by clicking on the lightbulb. Otherwise, I'm stuck with no lightbulb.

<p><img src="https://github.com/vfonic/vscode-ada-lightbulb/raw/master/resources/Lightbolb.gif" height="300px"></p>

**Comment by borislit** [comment]
@vfonic Gocha! Ill try to reproduce

**Comment by vfonic** [comment]
I do agree with @yyynnn, since this seems like something that would be much faster when available from the command menu. I've seen many other extensions easily add things to command menu, that are definitely context-only. Is there some way to give the user the possibility to choose whether or not he wants to be able to kick off this action from the command menu? Maybe some macro or something. I'm not 100% sure how it all works together under the hood.

**Comment by borislit** [comment]
I see what you mean. But wont it be weird to keep checking in the command pallet whether an option has appear or not?
Maybe what would be better is just allow configuring hot keys for each action? So lets say once u have selected your class component you'd just press `cmd + f` or something and wolla. WYDT?

**Comment by vfonic** [comment]
Yes! If that's possible, that would be perfect!

**Comment by borislit** [comment]
@vfonic a small updated - i haven't forgotten about that your suggestion with the hot keys and plan to include in the next release!

**Comment by iaarnio** [comment]
In my case I also had Eslint extension (dbaeumer.vscode-eslint) active and possibly it's lightbulb was overriding Glean lightbulb (as file had also linting problems).  

Glean lightbulb became visible after disabling Eslint extension. Sure in the long run I would like to keep them both enabled.

**Comment by Nantris** [comment]
It would be really nice if this was addressed after all these years. Glean does _NOTHING_ on my machine (besides lead me on wild goose chases for features that cannot be accessed.)
